### PR TITLE
Fix/search logic

### PR DESCRIPTION
**Pull request by m1073496**
## What does this PR do?
- Adds logic to search by genre and overview
- Fixes bug that was clearing search inputs when user clicks onto Movie Details view

## How should it be tested?
 - Open the app and make a search by genre or overview keyword
 - Click on movie to get to Movie Details view and then exit, search terms should still appear in search bar 
 - All previous functionality should remain

### CODE REVIEW

**Comment by m1073496** [comment]
[Logic for search by genre/overview](https://trello.com/c/fITc1gRW/48-logic-for-search-by-genre-overview)

**Comment by m1073496** [comment]
[Navigation w Search](https://trello.com/c/omt9ToG2/46-navigation-w-search)

File: src/app/App.js [hunk 1]
```diff
@@ -54,7 +57,14 @@ export default class App extends Component {
...
 
   filterMovies = (query) => {
-    const filteredList = this.state.movies.filter(movie => movie.title.toUpperCase().includes(query.toUpperCase()))
+    const formattedQuery = query.toUpperCase();
+
+    const filteredList = this.state.movies.filter(movie => {
+    return movie.title.toUpperCase().includes(formattedQuery)
+    || movie.genres.toString().toUpperCase().includes(formattedQuery)
+    || movie.overview.toUpperCase().includes(formattedQuery)
+  });
```
Comment (cbmackintosh) [inline src/app/App.js:66]: Good refactor here - added search bar capability but the code is a lot more readable.

File: src/container/MovieContainer.js [hunk 2]
```diff
@@ -17,9 +16,9 @@ const MovieContainer = ({ movies, filterMovies, isLoaded }) => {
   })
 
   if (isLoaded) {
+    document.querySelector('.search-bar').classList.remove('hidden')
     return (
```
Comment (cbmackintosh) [inline src/container/MovieContainer.js:20]: Did some more research on this and I found a few articles suggesting that using querySelectors in React is not best practice. We can look at other potential solutions if time allows but I'm not too worried about it. My idea was to try and save search queries in state to pass around between the components, but this seems much simpler and uses much less code.

**Review by cbmackintosh — APPROVED** [review]
Nice work figuring this one out. I think this resolves all navigational bugs on the app that I'm aware of.

**Pull request merged**